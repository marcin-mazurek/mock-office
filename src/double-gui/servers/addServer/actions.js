export const INIT = 'addServer/INIT';

export default (name, port, serverType, isSecure, keyPath, certPath) => ({
  type: INIT,
  name,
  port,
  serverType,
  isSecure,
  keyPath,
  certPath
});
