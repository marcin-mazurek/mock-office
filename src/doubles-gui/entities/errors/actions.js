export const ADD = 'errors/ADD';
export const REMOVE = 'errors/REMOVE';

export const add = reason => ({
  type: ADD,
  reason
});

export const remove = id => ({
  type: REMOVE,
  id
});
