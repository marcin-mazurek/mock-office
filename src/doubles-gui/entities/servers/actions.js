export const ADD = 'servers/ADD';
export const SELECT = 'servers/SELECT';
export const START = 'servers/START';
export const STOP = 'servers/STOP';
export const REMOVE = 'servers/REMOVE';
export const RENAME = 'servers/RENAME';

export const add = (name, port, serverType, id, secure) => ({
  type: ADD,
  name,
  port,
  id,
  serverType,
  secure
});

export const select = id => ({
  type: SELECT,
  id
});

export const start = id => ({
  type: START,
  id
});

export const stop = id => ({
  type: STOP,
  id
});

export const remove = id => ({
  type: REMOVE,
  id
});

export const rename = (id, name) => ({
  type: RENAME,
  id,
  name
});
