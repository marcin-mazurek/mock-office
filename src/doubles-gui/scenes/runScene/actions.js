export const RUN = 'scenes/RUN';
export const STOP = 'scenes/STOP';

export const run = (serverId, taskId) => ({
  type: RUN,
  serverId,
  taskId
});

export const stop = (serverId, taskId) => ({
  type: STOP,
  serverId,
  taskId
});
