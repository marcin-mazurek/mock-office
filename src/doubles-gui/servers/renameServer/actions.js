export const INIT = 'renameServer/INIT';

export const init = (id, name) => ({
  type: INIT,
  id,
  name
});
