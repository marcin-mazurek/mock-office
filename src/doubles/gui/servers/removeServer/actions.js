export const INIT = 'removeServer/INIT';

export const init = id => ({
  type: INIT,
  id
});
