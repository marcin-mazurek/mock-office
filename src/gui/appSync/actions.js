export const RESTORE_STATE = 'appSync/RESTORE_STATE';
export const restoreStateAction = servers => ({
  type: RESTORE_STATE,
  servers
});
