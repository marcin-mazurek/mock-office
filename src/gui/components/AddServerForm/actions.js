export const SUBMIT_BUTTON_CLICKED = 'addServerForm/SUBMIT_BUTTON_CLICKED';
export const submitButtonClickAction = values => ({
  type: SUBMIT_BUTTON_CLICKED,
  values
});
