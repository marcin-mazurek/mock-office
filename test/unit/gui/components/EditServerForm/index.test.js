import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  EditServerFormConnect,
  EditServerForm
} from '../../../../../src/gui/components/EditServerForm';
import configureStore from '../../../../../src/gui/app/configureStore';
import { succeededAction } from '../../../../../src/gui/epics/addServer/actions';

describe('EditServerForm', () => {
  test('default snapshot', () => {
    const wrapper = shallow(
      <EditServerForm
        handleSubmit={() => {
        }}
        serverId={'id'}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('EditServerFormConnect', () => {
  it('should be connected to store', () => {
    const store = configureStore();
    store.dispatch(succeededAction({
      data: {
        name: 'Awesome server',
        port: 3000,
        type: 'http',
        secure: false,
        scenario: 'cj45fwix00001yrve66ecdxp0',
        running: false,
        id: 'server-id'
      }
    }));
    const wrapper = mount(
      <Provider store={store}>
        <EditServerFormConnect serverId="server-id" />
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
