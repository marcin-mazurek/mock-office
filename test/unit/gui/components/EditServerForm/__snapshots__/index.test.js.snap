// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditServerForm default snapshot 1`] = `
<form
  action="#"
  className="form"
  onSubmit={[Function]}
>
  <div
    className="form__header"
  >
    <img
      alt=""
      className="form__header-icon"
      src=""
    />
    Edit server
  </div>
  <div
    className="form-row"
  >
    <div
      className="form__field"
    >
      <label
        className="form-field__label"
        htmlFor="server-name"
      >
        Name:
      </label>
      <Field
        className="form-field__input"
        component="input"
        name="name"
        type="text"
      />
    </div>
  </div>
  <div
    className="form-row"
  >
    <div
      className="form__col"
    >
      <div
        className="form__field"
      >
        <label
          className="form-field__label"
          htmlFor="server-port"
        >
          Port:
        </label>
        <Field
          className="form-field__input"
          component="input"
          name="port"
          type="number"
        />
      </div>
    </div>
  </div>
  <div
    className="form-row"
  >
    <button
      className="button form__button"
      type="submit"
    >
      Save
    </button>
    <Link
      className="button form__button"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/server/id"
    >
      Cancel
    </Link>
  </div>
</form>
`;

exports[`EditServerFormConnect should be connected to store 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <Connect(ReduxForm)
    serverId="server-id"
  >
    <ReduxForm
      initialValues={
        Object {
          "name": "Awesome server",
          "port": 3000,
        }
      }
      onSubmit={[Function]}
      serverId="server-id"
    >
      <Connect(Form(EditServerForm))
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="editServer"
        getFormState={[Function]}
        initialValues={
          Object {
            "name": "Awesome server",
            "port": 3000,
          }
        }
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        serverId="server-id"
        shouldAsyncValidate={[Function]}
        shouldValidate={[Function]}
        touchOnBlur={true}
        touchOnChange={false}
      >
        <Form(EditServerForm)
          anyTouched={false}
          array={
            Object {
              "insert": [Function],
              "move": [Function],
              "pop": [Function],
              "push": [Function],
              "remove": [Function],
              "removeAll": [Function],
              "shift": [Function],
              "splice": [Function],
              "swap": [Function],
              "unshift": [Function],
            }
          }
          arrayInsert={[Function]}
          arrayMove={[Function]}
          arrayPop={[Function]}
          arrayPush={[Function]}
          arrayRemove={[Function]}
          arrayRemoveAll={[Function]}
          arrayShift={[Function]}
          arraySplice={[Function]}
          arraySwap={[Function]}
          arrayUnshift={[Function]}
          asyncValidating={false}
          autofill={[Function]}
          blur={[Function]}
          change={[Function]}
          clearAsyncError={[Function]}
          clearSubmit={[Function]}
          clearSubmitErrors={[Function]}
          destroy={[Function]}
          destroyOnUnmount={true}
          dirty={false}
          dispatch={[Function]}
          enableReinitialize={false}
          focus={[Function]}
          forceUnregisterOnUnmount={false}
          form="editServer"
          getFormState={[Function]}
          initialValues={
            Object {
              "name": "Awesome server",
              "port": 3000,
            }
          }
          initialize={[Function]}
          initialized={true}
          invalid={false}
          keepDirtyOnReinitialize={false}
          onSubmit={[Function]}
          persistentSubmitErrors={false}
          pristine={true}
          pure={true}
          registerField={[Function]}
          registeredFields={
            Object {
              "name": Object {
                "count": 1,
                "name": "name",
                "type": "Field",
              },
              "port": Object {
                "count": 1,
                "name": "port",
                "type": "Field",
              },
            }
          }
          reset={[Function]}
          serverId="server-id"
          setSubmitFailed={[Function]}
          setSubmitSucceeded={[Function]}
          shouldAsyncValidate={[Function]}
          shouldValidate={[Function]}
          startAsyncValidation={[Function]}
          startSubmit={[Function]}
          stopAsyncValidation={[Function]}
          stopSubmit={[Function]}
          submit={[Function]}
          submitFailed={false}
          submitSucceeded={false}
          submitting={false}
          syncErrors={Object {}}
          syncWarnings={Object {}}
          touch={[Function]}
          touchOnBlur={true}
          touchOnChange={false}
          unregisterField={[Function]}
          untouch={[Function]}
          updateSyncErrors={[Function]}
          updateSyncWarnings={[Function]}
          valid={true}
          validExceptSubmit={true}
          values={
            Object {
              "name": "Awesome server",
              "port": 3000,
            }
          }
        >
          <EditServerForm
            anyTouched={false}
            array={
              Object {
                "insert": [Function],
                "move": [Function],
                "pop": [Function],
                "push": [Function],
                "remove": [Function],
                "removeAll": [Function],
                "shift": [Function],
                "splice": [Function],
                "swap": [Function],
                "unshift": [Function],
              }
            }
            asyncValidate={[Function]}
            asyncValidating={false}
            autofill={[Function]}
            blur={[Function]}
            change={[Function]}
            clearAsyncError={[Function]}
            clearSubmit={[Function]}
            clearSubmitErrors={[Function]}
            destroy={[Function]}
            dirty={false}
            dispatch={[Function]}
            form="editServer"
            handleSubmit={[Function]}
            initialValues={
              Object {
                "name": "Awesome server",
                "port": 3000,
              }
            }
            initialize={[Function]}
            initialized={true}
            invalid={false}
            onSubmit={[Function]}
            pristine={true}
            pure={true}
            reset={[Function]}
            serverId="server-id"
            submit={[Function]}
            submitFailed={false}
            submitSucceeded={false}
            submitting={false}
            touch={[Function]}
            untouch={[Function]}
            valid={true}
          >
            <form
              action="#"
              className="form"
              onSubmit={[Function]}
            >
              <div
                className="form__header"
              >
                <img
                  alt=""
                  className="form__header-icon"
                  src=""
                />
                Edit server
              </div>
              <div
                className="form-row"
              >
                <div
                  className="form__field"
                >
                  <label
                    className="form-field__label"
                    htmlFor="server-name"
                  >
                    Name:
                  </label>
                  <Field
                    className="form-field__input"
                    component="input"
                    name="name"
                    type="text"
                  >
                    <Connect(ConnectedField)
                      _reduxForm={
                        Object {
                          "anyTouched": false,
                          "array": Object {
                            "insert": [Function],
                            "move": [Function],
                            "pop": [Function],
                            "push": [Function],
                            "remove": [Function],
                            "removeAll": [Function],
                            "shift": [Function],
                            "splice": [Function],
                            "swap": [Function],
                            "unshift": [Function],
                          },
                          "arrayInsert": [Function],
                          "arrayMove": [Function],
                          "arrayPop": [Function],
                          "arrayPush": [Function],
                          "arrayRemove": [Function],
                          "arrayRemoveAll": [Function],
                          "arrayShift": [Function],
                          "arraySplice": [Function],
                          "arraySwap": [Function],
                          "arrayUnshift": [Function],
                          "asyncErrors": undefined,
                          "asyncValidate": [Function],
                          "asyncValidating": false,
                          "autofill": [Function],
                          "blur": [Function],
                          "change": [Function],
                          "clearAsyncError": [Function],
                          "clearSubmit": [Function],
                          "clearSubmitErrors": [Function],
                          "destroy": [Function],
                          "destroyOnUnmount": true,
                          "dirty": false,
                          "dispatch": [Function],
                          "enableReinitialize": false,
                          "error": undefined,
                          "focus": [Function],
                          "forceUnregisterOnUnmount": false,
                          "form": "editServer",
                          "getFormState": [Function],
                          "getValues": [Function],
                          "initialValues": Object {
                            "name": "Awesome server",
                            "port": 3000,
                          },
                          "initialize": [Function],
                          "initialized": false,
                          "invalid": false,
                          "keepDirtyOnReinitialize": false,
                          "onSubmit": [Function],
                          "persistentSubmitErrors": false,
                          "pristine": true,
                          "pure": true,
                          "register": [Function],
                          "registerField": [Function],
                          "registerInnerOnSubmit": [Function],
                          "registeredFields": undefined,
                          "reset": [Function],
                          "sectionPrefix": undefined,
                          "serverId": "server-id",
                          "setSubmitFailed": [Function],
                          "setSubmitSucceeded": [Function],
                          "shouldAsyncValidate": [Function],
                          "shouldValidate": [Function],
                          "startAsyncValidation": [Function],
                          "startSubmit": [Function],
                          "stopAsyncValidation": [Function],
                          "stopSubmit": [Function],
                          "submit": [Function],
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "submitting": false,
                          "syncErrors": Object {},
                          "syncWarnings": Object {},
                          "touch": [Function],
                          "touchOnBlur": true,
                          "touchOnChange": false,
                          "triggerSubmit": undefined,
                          "unregister": [Function],
                          "unregisterField": [Function],
                          "untouch": [Function],
                          "updateSyncErrors": [Function],
                          "updateSyncWarnings": [Function],
                          "valid": true,
                          "validExceptSubmit": true,
                          "values": Object {
                            "name": "Awesome server",
                            "port": 3000,
                          },
                          "warning": undefined,
                        }
                      }
                      className="form-field__input"
                      component="input"
                      name="name"
                      normalize={[Function]}
                      type="text"
                    >
                      <ConnectedField
                        _reduxForm={
                          Object {
                            "anyTouched": false,
                            "array": Object {
                              "insert": [Function],
                              "move": [Function],
                              "pop": [Function],
                              "push": [Function],
                              "remove": [Function],
                              "removeAll": [Function],
                              "shift": [Function],
                              "splice": [Function],
                              "swap": [Function],
                              "unshift": [Function],
                            },
                            "arrayInsert": [Function],
                            "arrayMove": [Function],
                            "arrayPop": [Function],
                            "arrayPush": [Function],
                            "arrayRemove": [Function],
                            "arrayRemoveAll": [Function],
                            "arrayShift": [Function],
                            "arraySplice": [Function],
                            "arraySwap": [Function],
                            "arrayUnshift": [Function],
                            "asyncErrors": undefined,
                            "asyncValidate": [Function],
                            "asyncValidating": false,
                            "autofill": [Function],
                            "blur": [Function],
                            "change": [Function],
                            "clearAsyncError": [Function],
                            "clearSubmit": [Function],
                            "clearSubmitErrors": [Function],
                            "destroy": [Function],
                            "destroyOnUnmount": true,
                            "dirty": false,
                            "dispatch": [Function],
                            "enableReinitialize": false,
                            "error": undefined,
                            "focus": [Function],
                            "forceUnregisterOnUnmount": false,
                            "form": "editServer",
                            "getFormState": [Function],
                            "getValues": [Function],
                            "initialValues": Object {
                              "name": "Awesome server",
                              "port": 3000,
                            },
                            "initialize": [Function],
                            "initialized": false,
                            "invalid": false,
                            "keepDirtyOnReinitialize": false,
                            "onSubmit": [Function],
                            "persistentSubmitErrors": false,
                            "pristine": true,
                            "pure": true,
                            "register": [Function],
                            "registerField": [Function],
                            "registerInnerOnSubmit": [Function],
                            "registeredFields": undefined,
                            "reset": [Function],
                            "sectionPrefix": undefined,
                            "serverId": "server-id",
                            "setSubmitFailed": [Function],
                            "setSubmitSucceeded": [Function],
                            "shouldAsyncValidate": [Function],
                            "shouldValidate": [Function],
                            "startAsyncValidation": [Function],
                            "startSubmit": [Function],
                            "stopAsyncValidation": [Function],
                            "stopSubmit": [Function],
                            "submit": [Function],
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "submitting": false,
                            "syncErrors": Object {},
                            "syncWarnings": Object {},
                            "touch": [Function],
                            "touchOnBlur": true,
                            "touchOnChange": false,
                            "triggerSubmit": undefined,
                            "unregister": [Function],
                            "unregisterField": [Function],
                            "untouch": [Function],
                            "updateSyncErrors": [Function],
                            "updateSyncWarnings": [Function],
                            "valid": true,
                            "validExceptSubmit": true,
                            "values": Object {
                              "name": "Awesome server",
                              "port": 3000,
                            },
                            "warning": undefined,
                          }
                        }
                        asyncValidating={false}
                        className="form-field__input"
                        component="input"
                        dirty={false}
                        dispatch={[Function]}
                        initial="Awesome server"
                        name="name"
                        normalize={[Function]}
                        pristine={true}
                        type="text"
                        value="Awesome server"
                      >
                        <input
                          className="form-field__input"
                          name="name"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onDragStart={[Function]}
                          onDrop={[Function]}
                          onFocus={[Function]}
                          type="text"
                          value="Awesome server"
                        />
                      </ConnectedField>
                    </Connect(ConnectedField)>
                  </Field>
                </div>
              </div>
              <div
                className="form-row"
              >
                <div
                  className="form__col"
                >
                  <div
                    className="form__field"
                  >
                    <label
                      className="form-field__label"
                      htmlFor="server-port"
                    >
                      Port:
                    </label>
                    <Field
                      className="form-field__input"
                      component="input"
                      name="port"
                      type="number"
                    >
                      <Connect(ConnectedField)
                        _reduxForm={
                          Object {
                            "anyTouched": false,
                            "array": Object {
                              "insert": [Function],
                              "move": [Function],
                              "pop": [Function],
                              "push": [Function],
                              "remove": [Function],
                              "removeAll": [Function],
                              "shift": [Function],
                              "splice": [Function],
                              "swap": [Function],
                              "unshift": [Function],
                            },
                            "arrayInsert": [Function],
                            "arrayMove": [Function],
                            "arrayPop": [Function],
                            "arrayPush": [Function],
                            "arrayRemove": [Function],
                            "arrayRemoveAll": [Function],
                            "arrayShift": [Function],
                            "arraySplice": [Function],
                            "arraySwap": [Function],
                            "arrayUnshift": [Function],
                            "asyncErrors": undefined,
                            "asyncValidate": [Function],
                            "asyncValidating": false,
                            "autofill": [Function],
                            "blur": [Function],
                            "change": [Function],
                            "clearAsyncError": [Function],
                            "clearSubmit": [Function],
                            "clearSubmitErrors": [Function],
                            "destroy": [Function],
                            "destroyOnUnmount": true,
                            "dirty": false,
                            "dispatch": [Function],
                            "enableReinitialize": false,
                            "error": undefined,
                            "focus": [Function],
                            "forceUnregisterOnUnmount": false,
                            "form": "editServer",
                            "getFormState": [Function],
                            "getValues": [Function],
                            "initialValues": Object {
                              "name": "Awesome server",
                              "port": 3000,
                            },
                            "initialize": [Function],
                            "initialized": false,
                            "invalid": false,
                            "keepDirtyOnReinitialize": false,
                            "onSubmit": [Function],
                            "persistentSubmitErrors": false,
                            "pristine": true,
                            "pure": true,
                            "register": [Function],
                            "registerField": [Function],
                            "registerInnerOnSubmit": [Function],
                            "registeredFields": undefined,
                            "reset": [Function],
                            "sectionPrefix": undefined,
                            "serverId": "server-id",
                            "setSubmitFailed": [Function],
                            "setSubmitSucceeded": [Function],
                            "shouldAsyncValidate": [Function],
                            "shouldValidate": [Function],
                            "startAsyncValidation": [Function],
                            "startSubmit": [Function],
                            "stopAsyncValidation": [Function],
                            "stopSubmit": [Function],
                            "submit": [Function],
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "submitting": false,
                            "syncErrors": Object {},
                            "syncWarnings": Object {},
                            "touch": [Function],
                            "touchOnBlur": true,
                            "touchOnChange": false,
                            "triggerSubmit": undefined,
                            "unregister": [Function],
                            "unregisterField": [Function],
                            "untouch": [Function],
                            "updateSyncErrors": [Function],
                            "updateSyncWarnings": [Function],
                            "valid": true,
                            "validExceptSubmit": true,
                            "values": Object {
                              "name": "Awesome server",
                              "port": 3000,
                            },
                            "warning": undefined,
                          }
                        }
                        className="form-field__input"
                        component="input"
                        name="port"
                        normalize={[Function]}
                        type="number"
                      >
                        <ConnectedField
                          _reduxForm={
                            Object {
                              "anyTouched": false,
                              "array": Object {
                                "insert": [Function],
                                "move": [Function],
                                "pop": [Function],
                                "push": [Function],
                                "remove": [Function],
                                "removeAll": [Function],
                                "shift": [Function],
                                "splice": [Function],
                                "swap": [Function],
                                "unshift": [Function],
                              },
                              "arrayInsert": [Function],
                              "arrayMove": [Function],
                              "arrayPop": [Function],
                              "arrayPush": [Function],
                              "arrayRemove": [Function],
                              "arrayRemoveAll": [Function],
                              "arrayShift": [Function],
                              "arraySplice": [Function],
                              "arraySwap": [Function],
                              "arrayUnshift": [Function],
                              "asyncErrors": undefined,
                              "asyncValidate": [Function],
                              "asyncValidating": false,
                              "autofill": [Function],
                              "blur": [Function],
                              "change": [Function],
                              "clearAsyncError": [Function],
                              "clearSubmit": [Function],
                              "clearSubmitErrors": [Function],
                              "destroy": [Function],
                              "destroyOnUnmount": true,
                              "dirty": false,
                              "dispatch": [Function],
                              "enableReinitialize": false,
                              "error": undefined,
                              "focus": [Function],
                              "forceUnregisterOnUnmount": false,
                              "form": "editServer",
                              "getFormState": [Function],
                              "getValues": [Function],
                              "initialValues": Object {
                                "name": "Awesome server",
                                "port": 3000,
                              },
                              "initialize": [Function],
                              "initialized": false,
                              "invalid": false,
                              "keepDirtyOnReinitialize": false,
                              "onSubmit": [Function],
                              "persistentSubmitErrors": false,
                              "pristine": true,
                              "pure": true,
                              "register": [Function],
                              "registerField": [Function],
                              "registerInnerOnSubmit": [Function],
                              "registeredFields": undefined,
                              "reset": [Function],
                              "sectionPrefix": undefined,
                              "serverId": "server-id",
                              "setSubmitFailed": [Function],
                              "setSubmitSucceeded": [Function],
                              "shouldAsyncValidate": [Function],
                              "shouldValidate": [Function],
                              "startAsyncValidation": [Function],
                              "startSubmit": [Function],
                              "stopAsyncValidation": [Function],
                              "stopSubmit": [Function],
                              "submit": [Function],
                              "submitFailed": false,
                              "submitSucceeded": false,
                              "submitting": false,
                              "syncErrors": Object {},
                              "syncWarnings": Object {},
                              "touch": [Function],
                              "touchOnBlur": true,
                              "touchOnChange": false,
                              "triggerSubmit": undefined,
                              "unregister": [Function],
                              "unregisterField": [Function],
                              "untouch": [Function],
                              "updateSyncErrors": [Function],
                              "updateSyncWarnings": [Function],
                              "valid": true,
                              "validExceptSubmit": true,
                              "values": Object {
                                "name": "Awesome server",
                                "port": 3000,
                              },
                              "warning": undefined,
                            }
                          }
                          asyncValidating={false}
                          className="form-field__input"
                          component="input"
                          dirty={false}
                          dispatch={[Function]}
                          initial={3000}
                          name="port"
                          normalize={[Function]}
                          pristine={true}
                          type="number"
                          value={3000}
                        >
                          <input
                            className="form-field__input"
                            name="port"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onDragStart={[Function]}
                            onDrop={[Function]}
                            onFocus={[Function]}
                            type="number"
                            value={3000}
                          />
                        </ConnectedField>
                      </Connect(ConnectedField)>
                    </Field>
                  </div>
                </div>
              </div>
              <div
                className="form-row"
              >
                <button
                  className="button form__button"
                  type="submit"
                >
                  Save
                </button>
                <Link
                  className="button form__button"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/server/server-id"
                >
                  <a
                    className="button form__button"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Cancel
                  </a>
                </Link>
              </div>
            </form>
          </EditServerForm>
        </Form(EditServerForm)>
      </Connect(Form(EditServerForm))>
    </ReduxForm>
  </Connect(ReduxForm)>
</Provider>
`;
